### DISCUSSION TITLE
How do I set env: ['browser', 'node'] for new config?

### DISCUSSION THREAD

**Discussion by chiawendt** [eslint/eslint#16975] [Q&A] [answered]
I need globals like window and localStorage. How do I set it in new config?

**Comment by aladdin-add** [comment, marked as answer]
you need to set `languageOptions.globals`:

```js
export default [{
languageOptions: {
  globals: {...}
}
}];
```

**Comment by TheJaredWilcurt** [comment]
## ESLint 8

`eslintrc.cjs`

```js
module.exports = {
  env: {
    browser: true,
    node: true
  }
};
```

## ESLint 9

`eslint.config.js`

```js
import globals from 'globals';

export default [
  {
    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.nodeBuiltin
      }
    }
  }
];
```

## Notes

* You do not need to `npm install globals` because it is already a dependency of ESLint.
  * `eslint@9` > `@eslint/eslintrc@3` > `globals@14`
* Use `...globals.nodeBuiltin` if your project is ESM (uses `import` instead of `require`).
* Use `...globals.node` if your project is CJS (uses `require` instead of `import`).
* More info: https://www.npmjs.com/package/globals
